scripts/v2/clean-orphan-role-assignments.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+# This script exists to manually clean up the CI subscription role assignments if there get to be too many.
+# We could run it automatically but they're leaked relatively rarely and only clutter the UI (not a security issue as the
+# backing identity is deleted).
+# We should probably figure out what's creating the assignments and update that to do a best effort delete...
+# It's probably make-mi-fic.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+DELETE=0
+
+print_usage() {
+  echo "Usage: cleanup-orphan-role-assignments.sh [-y]"
+  echo "    -y: Actually delete everything. If not specified, the default is dry-run mode"
+}
+
+while getopts 'y' flag; do
+  case "${flag}" in
+    y) DELETE=1 ;;
+    *) print_usage
+       exit 1 ;;
+  esac
+done
+
+# This lists all of the role assignments with Owner and no principal - in our case that's role assignments f
+IDS=$(az role assignment list --query "([?(roleDefinitionName == 'Owner') && (principalName == '')].id).join(' ', [*])" | tr -d '"')
```

Comment:
Maybe worth adding a filter for 'Contributor'?

Reply:
Fixed this and also fixed at least one leak of roleassignments as well.